readthedocs/search/parsers.py

```diff
@@ -111,21 +111,16 @@ def _get_main_node(self, html):
           so they are children of the same parent node.
         - Return the body element itself if all checks above fail.
         """
-        body = html.body
-        main_node = body.css_first("[role=main]")
-        if main_node:
-            return main_node
+        from readthedocs.projects.models import AddonsConfig
 
-        main_node = body.css_first("main")
+        body = html.body
+        main_node = body.css_first(
+            self.project.addons.options_doctool_root_selector
+            or AddonsConfig.DEFAULT_ROOT_SELECTOR
+        )
         if main_node:
             return main_node
 
-        # TODO: this could be done in smarter way,
-        # checking for common parents between all h nodes.
-        first_header = body.css_first("h1")
```

Comment:
Another place we're losing this h1 logic as a fallback. Seems important, since otherwise we're changing how it works again.